### PR TITLE
David/fixes split data fit

### PR DESCRIPTION
**Pull request by DavidSarlle**
This branch contins some fixes for PR #279. 

Mostly fixes:
- Lorentzian fit failures
- Flipping fit
- Drag tunning fit
- some minor changes in plotting

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by Edoardo-Pedicillo — APPROVED** [review]

File: src/qibocal/protocols/characterization/allxy/allxy.py [hunk 1]
```diff
@@ -84,30 +85,34 @@ def _acquisition(
...
+                z_proj = 2 * results[ro_pulse.serial].ground_state_probability - 1
+                # store the results
+                r = {
+                    "probability": z_proj,
+                    "gateNumber": gateNumber,
+                    "beta_param": params.beta_param,
+                    "qubit": ro_pulse.qubit,
+                }
+                data.add(r)
+            count += 1
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/allxy/allxy.py:105]: `count` can be removed I think

File: src/qibocal/protocols/characterization/allxy/allxy.py [hunk 2]
```diff
@@ -84,30 +85,34 @@ def _acquisition(
...
+            for ro_pulse in ro_pulses.values():
+                z_proj = 2 * results[ro_pulse.serial].ground_state_probability - 1
+                # store the results
+                r = {
+                    "probability": z_proj,
+                    "gateNumber": gateNumber,
+                    "beta_param": params.beta_param,
+                    "qubit": ro_pulse.qubit,
+                }
+                data.add(r)
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/allxy/allxy.py:104]: Here I believe that the indentation was corrected before. Did you test it?

File: src/qibocal/protocols/characterization/utils.py [hunk 3]
```diff
@@ -77,7 +77,31 @@ def lorentzian_fit(data, qubit):
...
+            "fitted_parameters": fitted_parameters,
+            "bare_frequency": bare_frequency,
+            "amplitude": amplitudes,
+        }
+    else:
+        return {
+            "frequency": frequency,
+            "fitted_parameters": fitted_parameters,
+            "amplitude": amplitudes,
+        }
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/utils.py:104]: I think that this part is not needed.
It should already be in `_fit` of `resonator_spectroscopy`.

**Review by andrea-pasquale — CHANGES_REQUESTED** [review]
Thanks @DavidSarlle for implementing this.
Please find below a few comments, nothing critical just some minor changes.

**Review by andrea-pasquale — CHANGES_REQUESTED** [review]
Thanks @DavidSarlle for implementing this.
Please find below a few comments, nothing critical just some minor changes.

File: runcards/actions_qq.yml
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply runcards/actions_qq.yml:1]: Done

File: runcards/autocalibration_qblox.yml
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply runcards/autocalibration_qblox.yml:1]: Done

File: src/qibocal/protocols/characterization/allxy/allxy_drag_pulse_tuning.py [hunk 4]
```diff
@@ -62,6 +62,8 @@ def _acquisition(
 
     data = AllXYDragData()
 
+    # repeat the experiment as many times as defined by software_averages
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/allxy/allxy_drag_pulse_tuning.py:65]: Done

File: src/qibocal/protocols/characterization/classification.py [hunk 5]
```diff
@@ -105,6 +109,7 @@ def _acquisition(
             {
                 "qubit": [ro_pulse.qubit] * params.nshots,
                 "state": [0] * params.nshots,
+                "nshots": [params.nshots] * params.nshots,
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:112]: removed

File: src/qibocal/protocols/characterization/classification.py [hunk 6]
```diff
@@ -121,6 +126,7 @@ def _acquisition(
             {
                 "qubit": [ro_pulse.qubit] * params.nshots,
                 "state": [1] * params.nshots,
+                "nshots": [params.nshots] * params.nshots,
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:129]: removed

File: src/qibocal/protocols/characterization/classification.py [hunk 7]
```diff
@@ -130,14 +136,15 @@ def _acquisition(
...
     fidelities, assignment_fidelities = {}, {}
-    mean_gnd_state = {}
-    mean_exc_state = {}
+    mean_gnd_states = {}
+    mean_exc_states = {}
 
     for qubit in qubits:
         qubit_data = data.df[data.df["qubit"] == qubit].drop(
-            columns=["qubit", "MSR", "phase"]
+            columns=["qubit", "MSR", "phase", "nshots"]
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:147]: removed

File: src/qibocal/protocols/characterization/classification.py [hunk 8]
```diff
@@ -319,12 +324,12 @@ def _plot(
...
-        + f"{qubit} | Rotation Angle: {fit.rotation_angle[qubit]:.3f} rad <br>"
-        + f"{qubit} | Threshold: {fit.threshold[qubit]:.4f} <br>"
-        + f"{qubit} | Fidelity: {fit.fidelity[qubit]:.3f} <br>"
-        + f"{qubit} | Assignment Fidelity: {fit.assignment_fidelity[qubit]:.3f} <br>"
+        + f"q{qubit} | average state 0: {fit.mean_gnd_states[qubit]:.7f} <br>"
+        + f"q{qubit} | average state 1: {fit.mean_exc_states[qubit]:.7f} <br>"
+        + f"q{qubit} | rotation_angle: {fit.rotation_angle[qubit]:.3f} rad <br>"
+        + f"q{qubit} | threshold: {fit.threshold[qubit]:.6f} <br>"
+        + f"q{qubit} | fidelity: {fit.fidelity[qubit]:.3f} <br>"
+        + f"q{qubit} | assignment fidelity: {fit.assignment_fidelity[qubit]:.3f} <br>"
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:332]: Done

File: src/qibocal/protocols/characterization/flipping.py [hunk 9]
```diff
@@ -225,8 +231,8 @@ def _plot(data: FlippngData, fit: FlippingResults, qubit):
             ),
         )
         fitting_report = fitting_report + (
-            f"{qubit}| Amplitude Correction Factor: {fit.amplitude[qubit][0]:.4f}<br>"
-            + f"{qubit} | Corrected Amplitude: {fit.amplitude_factors[qubit]:.4f}<br><br>"
+            f"q{qubit} | amplitude_correction_factor: {fit.amplitude_factors[qubit]:.4f}<br>"
+            + f"q{qubit} | corrected_amplitude: {fit.amplitudes[qubit][0]:.4f}<br><br>"
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/flipping.py:235]: Done

File: src/qibocal/protocols/characterization/classification.py [hunk 10]
```diff
@@ -183,24 +190,22 @@ def _fit(data: SingleShotClassificationData) -> SingleShotClassificationResults:
...
+        assignment_fidelity = 1 - (errors_state1 + errors_state0) / nshots / 2
         thresholds[qubit] = threshold
-        rotation_angles[
-            qubit
-        ] = -rotation_angle  # TODO: qblox driver np.rad2deg(-rotation_angle)
-        fidelities[qubit] = fidelity
-        mean_gnd_state[qubit] = iq_mean_state0
-        mean_exc_state[qubit] = iq_mean_state1
-        assignment_fidelities[qubit] = assignment_fidelity
+        rotation_angles[qubit] = rotation_angle
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:198]: I removed from here and changed in the driver on update the runcard. But better to do it here.

[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/classification.py:198]: Fixed

File: src/qibocal/protocols/characterization/utils.py (hunk 3 above)
Comment (DavidSarlle) [reply src/qibocal/protocols/characterization/utils.py:104]: At the very beggining was in inside the lorentzian_fit. Sorry, I did not update after someone change it. 
Fixed

**Pull request merged**